packages/gafl-webapp-service/src/pages/multibuy/remove-licence/__tests__/update-transaction.spec.js

```diff
@@ -76,79 +74,96 @@ describe('remove-licence > update transaction', () => {
     await updateTransaction(mockRequest)
-    expect(mockRequest.cache().helpers.status.set()).toEqual(expect.objectContaining(status))
+    expect(statusSet).toHaveBeenCalledWith(status)
   })
 
-  it('addressLookup.set is being called with a permission removed', async () => {
+  it('addressLookup cache is updated after permission is deleted', async () => {
     const addressLookup = { permissions: [{}, {}, {}] }
-    const mockRequest = createMockRequest({ cache: { addressLookup: addressLookup } })
+    const mockRequest = getMockRequest({ cache: { addressLookup: addressLookup } })
     await updateTransaction(mockRequest)
     expect(mockRequest.cache().helpers.addressLookup.set()).toEqual(expect.objectContaining(addressLookup))
   })
 
   describe('after remove licence', () => {
-    it('transaction does not contain the transaction currentPermission', async () => {
+    it('transaction cache has a permission removed', async () => {
       const transaction = { permissions: [getTransactionPermissionOne(), getTransactionPermissionTwo(), getTransactionPermissionThree()] }
-      const mockRequest = createMockRequest({ cache: { transaction: transaction } })
+      const mockRequest = getMockRequest({ cache: { transaction: transaction } })
       await updateTransaction(mockRequest)
       expect(transaction.permissions.every(p => p.licenceType === 'trout-and-coarse')).toBeTruthy()
     })
 
-    it('page does not contain the page currentPermission', async () => {
+    it('page cache has a permission removed', async () => {
       const page = { permissions: [getPagePermission({ [REMOVE_LICENCE.page]: true }), getPagePermission(), getPagePermission()] }
-      const mockRequest = createMockRequest({ cache: { page: page } })
+      const mockRequest = getMockRequest({ cache: { page: page } })
       await updateTransaction(mockRequest)
       expect(page.permissions.length).toEqual(2)
     })
 
-    it('status does not contain the status currentPermission', async () => {
+    it('status cache has a permission removed', async () => {
       const status = { permissions: [getStatusPermission({ [REMOVE_LICENCE.page]: true }), getStatusPermission(), getStatusPermission()] }
-      const mockRequest = createMockRequest({ cache: { status: status } })
+      const mockRequest = getMockRequest({ cache: { status: status } })
       await updateTransaction(mockRequest)
       expect(status.permissions.length).toEqual(2)
     })
 
-    it('addressLookup does not contain the addressLookup currentPermission', async () => {
+    it('addressLookup cache has a permission removed', async () => {
       const addressLookup = { permissions: [{}, {}, {}] }
-      const mockRequest = createMockRequest({ cache: { addressLookup: addressLookup } })
+      const mockRequest = getMockRequest({ cache: { addressLookup: addressLookup } })
       await updateTransaction(mockRequest)
       expect(addressLookup.permissions.length).toEqual(2)
```

Comment:
This, and the preceding two tests, are just checking the length. So, to pass the test, the executing code could just delete a random value. You need to test that the exact specified object was removed